# Step 1: Data Exploration

Goal: Understand the dataset before making any changes.

Rule: Never modify data during exploration.

In [3]:
import pandas as pd

In [4]:
df = pd.read_csv("developer_ai_learning_raw.csv")

In [5]:
df.shape

(49191, 11)

In [ ]:
df.head()

In [ ]:
df.dtypes

In [ ]:
df.isna().sum()

In [ ]:
df.duplicated().sum()

In [7]:
df['ResponseId'].duplicated().sum()

np.int64(0)

In [ ]:
df['Age'].value_counts()

In [ ]:
df['AISelect'].value_counts(dropna=False)

In [6]:
df['LearnCode'].head()

0    Online Courses or Certification (includes all ...
1    Online Courses or Certification (includes all ...
2    Online Courses or Certification (includes all ...
3    Other online resources (e.g. standard search, ...
4                                                  NaN
Name: LearnCode, dtype: str

## Summary

What we found:
- 49,191 rows, 11 columns
- No duplicates
- ResponseId is unique
- YearsCode is float (has missing values)
- LearnCode and AILearnHow contain semicolons

Actions needed:
1. Convert YearsCode to Int64
2. Split LearnCode and AILearnHow
3. Keep missing values as null